ldap/src/main/java/org/springframework/security/ldap/server/ApacheDSContainer.java

```diff
@@ -143,6 +148,9 @@ public void afterPropertiesSet() throws Exception {
 		server.setDirectoryService(service);
 		// AbstractLdapIntegrationTests assume IPv4, so we specify the same here
 
+		if (this.port == 0) {
+			this.port = getRandomPort();
+		}
```

Comment:
Sorry for the confusion. We are already finding a random port in this way within our configuration support. 

The problem is sometimes the port gets resolved to an open port and assigned, the port is then taken up by another process or thread, and then this code tries to use the now taken port.

To fix this we want to be able to pass in `0` to `TcpTransport` to avoid a race condition.

Reply:
Thanks for the review. Do I understand correctly that we should also add an accessor to `ApacheDSContainer` for configured port value i.e `0`?

Reply:
It's not for the configured value, but for the actual port that is resolved by `TcpTransport`.

Reply:
I think it should be `getLocalPort` for the actual port.

Reply:
Yes we want the name of the method to be getLocalPort

Reply:
Done.